lib/dns/server.js

```diff
@@ -321,15 +323,35 @@ class RootServer extends DNSServer {
       // TLD '._synth' is being queried on its own, send SOA
       // so recursive asks again with complete synth record.
       if (labels.length === 1) {
+        // Empty non-terminal proof:
+        res.authority.push(nsec.create('_synth.',
+          '\\000._synth.', TYPE_MAP_EMPTY));
+        key.signZSK(res.authority, types.NSEC);
+
         res.authority.push(this.toSOA());
         key.signZSK(res.authority, types.SOA);
+
         return res;
       }
 
       const hash = util.label(name, labels, -2);
       const ip = IP.map(base32.decodeHex(hash.substring(1)));
+      const synthType = IP.isIPv4(ip) ? types.A : types.AAAA;
+
+      // Query must be for the correct synth version
+      if (type !== synthType) {
+        // SYNTH4/6 proof:
+        const typeMap = synthType === types.A ? TYPE_MAP_A : TYPE_MAP_AAAA;
+        res.authority.push(nsec.create(name, '\\000.'+name, typeMap));
```

Comment:
```suggestion
        res.authority.push(nsec.create(name, '\\000.' + name, typeMap));
```

nit: spaces around `+`